src/Lang/CorePriv/SExprPrinter.ml

```diff
@@ -28,6 +28,9 @@ let tr_var  x = Sym (Var.unique_name x)
 let tr_tvar (x : _ tvar) =
   Sym (Printf.sprintf "tp%s" (UID.to_string x.uid))
 
+let tr_uvar (x : UID.t) =
+  Sym (Printf.sprintf "weak%s" (UID.to_string x))
```

Comment:
It should be `weak#42` or `uvar#42`? At the beginning I thought that `uvar#42` would be better, but now I'm not convinced.